### PR TITLE
Sparse_Matrix_Representation.cpp

### PR DESCRIPTION
**Pull request by TEJASWI-TEJASWI**
This is an optimized solution of Sparse Matrix .
Sparse Matrix having more number of zeroes elements.

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

#573 

Yes
### Description

A sparse matrix is a two-dimensional data object made up of m rows and n columns. Most of the elements in the Sparse matrix are 0. A sparse matrix is preferred over the simple matrix as it takes less space than a simple matrix. Sparse Representation using array traverse only to the non-zero element which in return reduce both time and space complexity and even make the code more efficient.

Space Complexity: O(1)
### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#573

### CODE REVIEW

**Comment by TEJASWI-TEJASWI** [comment]
Kindly review this PR @Kajol-Kumari @SKAUL05 @iamrajiv @plazzy99 @atarax665

**Comment by TEJASWI-TEJASWI** [comment]
Kindly review this PR @Kajol-Kumari @SKAUL05 @iamrajiv @plazzy99 @atarax665

**Comment by vatsalkesarwani12** [comment]
check the checklist

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+        else
+            cout<<"0 ";
+        }
+        cout<<endl;
+    }
+    return os;
+}
+int main()
+{
+    Sparse s1(5,5,5);
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp]: take dynamic input

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+    Sparse s2(5,5,5);
+    cin>>s1;
+    cin>>s2;
+    Sparse sum=s1+s2;
+    cout<<"First Matrix"<<endl<<s1;
+    cout<<"Second Matrix"<<endl<<s2;
+    cout<<"Sum Matrix"<<endl<<sum;
+    return 0;
+}
```
Comment (vatsalkesarwani12) [inline C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:153]: sample input and output in comment

**Review by vatsalkesarwani12 — CHANGES_REQUESTED** [review]

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+//of the matrix have 0 value, then it is called a sparse matrix.
+#include <iostream>
+using namespace std;
+class Element
+{
+public:
+    int i;
+    int j;
+    int x;
+};
```
Comment (Abhijit2505) [inline C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:14]: Please change the variables names to proper words instead of just alphabets.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Please change your variable names into suitable ones.

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp (hunk 2 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:153]: I resolve this issue please have a look @plazzy99

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp (hunk 3 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:14]: I resolve this issue please have a look @Abhijit2505

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp (hunk 2 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:153]: I resolve this issue please have a look @plazzy99

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp (hunk 3 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp:14]: I resolve this issue please have a look @Abhijit2505

**Review by TEJASWI-TEJASWI — COMMENTED** [review]
I resolve the issue assigned by  @Abhijit2505 and @plazzy99.
please have a look @Abhijit2505

File: C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp (hunk 1 above)
Comment (TEJASWI-TEJASWI) [reply C-Plus-Plus/ds/Sparse_Matrix_Representation.cpp]: I resolve this issue please have a look @plazzy99

**Comment by TEJASWI-TEJASWI** [comment]
I resolve this issue please have a look @Abhijit2505 and @plazzy99 @SKAUL05  @atarax665

**Review by vatsalkesarwani12 — APPROVED** [review]

**Comment by TEJASWI-TEJASWI** [comment]
I resolve this issue please have a look @Abhijit2505 and @plazzy99 @SKAUL05 @atarax665

**Review by Abhijit2505 — APPROVED** [review]

**Pull request merged**